onnxruntime/python/tools/transformers/fusion_attention_clip.py

```diff
@@ -183,14 +193,27 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
             if qk_nodes is None:
                 qk_nodes = self.model.match_parent_path(matmul_qkv, ["Softmax", "Add", "Mul", "MatMul"], [0, 0, 0, 0])
                 if qk_nodes is None:
-                    qk_nodes = self.model.match_parent_path(
-                        matmul_qkv, ["Cast", "Cast", "Softmax", "Add", "Mul", "MatMul"], [0, 0, 0, 0, 0, 0]
-                    )
+                    # If attention mask is not used, we can still match the qk path.
```

Comment:
Suggest to change the condition so that layout is more friendly:
Before:
```
if qk_nodes is None:
    ...
else:
   add_mask = qk_nodes[1]
```

To

```
if qk_nodes is not None:
   add_mask = qk_nodes[1]
else:
   ...
```

Another possible change is to use [match_parent_paths](https://github.com/microsoft/onnxruntime/blob/24620e70d9f14956a0dc84bb8a332dcd64c95a94/onnxruntime/python/tools/quantization/fusions/fusion.py#L276).